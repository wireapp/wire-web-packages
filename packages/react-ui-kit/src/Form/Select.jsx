/*
 * Wire
 * Copyright (C) 2017 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import {COLOR} from '../Identity';
import {Input} from './Input';
import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

const StyledContainerSelect = styled.div`
  /* appearance */
  /* positioning */
  display: flex;
  margin-top: 3px;
  overflow: hidden;
  position: relative;

  svg {
    pointer-events: none;
    position: absolute;
    right: 8px;
    top: 22px;
  }
`;

const StyledSelect = Input.withComponent('select').extend`
  /* appearance */
  background: ${COLOR.GRAY_LIGHTEN_92};
  background-image: none;
  border: none;
  box-shadow: none;
  color: ${COLOR.GRAY_DARKEN_48};
  font-weight: 300;
  overflow: hidden;
  -moz-appearance: none;
  -webkit-appearance: none;

  /* positioning */
  margin-top: 3px;
  padding: 0 20px 0 12px;
  width: 100%;

  &:focus {
    /* appearance */
    outline: none;
  }
  &:disabled {
    /* appearance */
    color: ${COLOR.GRAY};
    + svg {
      /* appearance */
      display: none;
    }
  }
`;

const ArrowDown = () => (
  <svg width="8" height="8" viewBox="0 0 8 8">
    <path fill={COLOR.GRAY_DARKEN_48} fillRule="evenodd" d="M0 2h8L4 7" />
  </svg>
);

const Select = ({children, ...props}) => {
  return (
    <StyledContainerSelect>
      <StyledSelect {...props}>{children}</StyledSelect>
      <ArrowDown />
    </StyledContainerSelect>
  );
};

Select.propTypes = {
  ...Input.propTypes,
  children: PropTypes.node,
};

Select.defaultProps = {
  children: null,
};

export {Select};
