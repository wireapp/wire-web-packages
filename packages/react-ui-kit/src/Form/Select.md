Demo:

```js
import {Fragment} from 'react';
import {Columns, Column, Select} from '@wireapp/react-ui-kit';

<Fragment>
  <Columns>
    <Column>Select</Column>
    <Column>
      <Select>
        <option>a</option>
        <option>b</option>
        <option>c</option>
        <option>d</option>
      </Select>
    </Column>
  </Columns>
  <Columns>
    <Column>Labeled Select</Column>
    <Column>
      <Select label="Labeled Select">
        <option>a</option>
        <option>b</option>
        <option>c</option>
        <option>d</option>
      </Select>
    </Column>
  </Columns>
  <Columns>
    <Column>Disabled Select</Column>
    <Column>
      <Select disabled>
        <option>a</option>
      </Select>
    </Column>
  </Columns>
  <Columns>
    <Column>Invalid Select</Column>
    <Column>
      <Select markInvalid>
        <option>a</option>
      </Select>
    </Column>
  </Columns>
</Fragment>;
```
