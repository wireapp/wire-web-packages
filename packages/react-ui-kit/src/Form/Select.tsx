/*
 * Wire
 * Copyright (C) 2018 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

/** @jsx jsx */
import {ObjectInterpolation, jsx} from '@emotion/core';
import {COLOR} from '../Identity';
import {inlineSVG} from '../util';
import {InputProps, inputStyle} from './Input';

export interface SelectProps<T = HTMLSelectElement> extends InputProps<T> {
  disabled?: boolean;
}

const ArrowDown = `
  <svg xmlns="http://www.w3.org/2000/svg" width="8" height="8" viewBox="0 0 8 8">
    <path fill="${COLOR.TEXT}" fillRule="evenodd" d="M0 2h8L4 7" />
  </svg>
`;

const selectStyle: (props: SelectProps) => ObjectInterpolation<undefined> = ({disabled = false, ...props}) => ({
  ...inputStyle(props as any),
  '&:-moz-focusring': {
    color: 'transparent',
    textShadow: '0 0 0 #000',
  },
  '&:disabled': {
    color: COLOR.GRAY,
  },
  appearance: 'none',
  background: disabled
    ? COLOR.shade(COLOR.WHITE, 0.06)
    : `${COLOR.WHITE} center right 16px no-repeat url("${inlineSVG(ArrowDown)}")`,
  cursor: disabled ? 'normal' : 'pointer',
  fontWeight: 300,
  paddingRight: '30px',
});

const Select = (props: SelectProps) => <select css={selectStyle(props)} {...props} />;

export {Select, selectStyle};
