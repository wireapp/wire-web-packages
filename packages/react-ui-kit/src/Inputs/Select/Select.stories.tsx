/*
 * Wire
 * Copyright (C) 2025 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import type {Meta, StoryObj} from '@storybook/react';

import {Select} from './Select';

const options = [
  {value: '1', label: 'Option 1'},
  {value: '2', label: 'Option 2'},
  {value: '3', label: 'Option 3'},
];

const meta = {
  title: 'Inputs/Select',
  component: Select,
  parameters: {
    layout: 'centered',
  },
  decorators: [
    Story => (
      <div style={{width: '350px'}}>
        <Story />
      </div>
    ),
  ],
  tags: ['autodocs'],
} satisfies Meta<typeof Select>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    placeholder: 'Select an option',
    id: 'default-select',
    dataUieName: 'default-select',
    options,
  },
};

export const WithLabel: Story = {
  args: {
    placeholder: 'Select an option',
    id: 'select-example',
    dataUieName: 'select-with-label',
    options,
    label: 'Choose an option',
  },
};

export const Invalid: Story = {
  args: {
    placeholder: 'Select an option',
    id: 'invalid-select',
    dataUieName: 'invalid-select',
    options,
    markInvalid: true,
  },
};

export const Disabled: Story = {
  args: {
    placeholder: 'Select an option',
    id: 'disabled-select',
    dataUieName: 'disabled-select',
    options,
    disabled: true,
  },
};
