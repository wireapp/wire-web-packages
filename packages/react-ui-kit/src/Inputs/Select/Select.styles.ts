/*
 * Wire
 * Copyright (C) 2022 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import {CSSObject} from '@emotion/react';
import {GroupBase, StylesConfig} from 'react-select';

import {
  baseContainerStyles,
  baseControlStyles,
  baseDropdownIndicatorStyles,
  baseIndicatorSeparatorStyles,
  baseMenuListStyles,
  baseMenuStyles,
  baseOptionStyles,
} from './BaseSelect/BaseSelect.styles';
import {Option} from './Select';
import {isGroup} from './SelectOption/SelectOption';

import {Theme} from '../../Identity/Theme';

interface CustomStylesParams {
  theme: Theme;
  markInvalid?: boolean;
  menuPosition?: 'absolute' | 'relative';
  controlCSS: CSSObject;
  containerCSS: CSSObject;
  menuCSS: CSSObject;
  groupCSS: CSSObject;
  groupHeadingCSS: CSSObject;
}

export const customStyles = ({
  theme,
  markInvalid = false,
  menuPosition = 'absolute',
  controlCSS,
  containerCSS,
  menuCSS,
  groupCSS,
  groupHeadingCSS,
}: CustomStylesParams): StylesConfig<Option, false, GroupBase<Option>> => ({
  indicatorSeparator: baseIndicatorSeparatorStyles,
  indicatorsContainer: provided => provided,
  control: (_provided, {isDisabled, selectProps}) =>
    baseControlStyles({theme, isDisabled, markInvalid, selectProps, controlCSS}),
  dropdownIndicator: (provided, selectProps) => ({
    ...provided,
    ...baseDropdownIndicatorStyles({theme, selectProps}),
  }),
  container: (_, {options}) =>
    isGroup(options)
      ? {
          '& > div': {
            display: 'inline',
            position: 'relative',
            top: '-10px',
            ...containerCSS,
          },
        }
      : baseContainerStyles(containerCSS),
  menu: (provided, {options}) => ({
    ...provided,
    ...baseMenuStyles({theme, menuPosition}),
    ...(isGroup(options) && {
      minWidth: '400px',
    }),
    ...menuCSS,
  }),
  menuList: provided => ({
    ...provided,
    ...baseMenuListStyles(),
  }),
  option: (provided, {isMulti, isDisabled, isFocused, isSelected, options, data}) => ({
    ...provided,
    ...baseOptionStyles({theme, isMulti, isDisabled, isFocused, isSelected}),
    padding: isGroup(options) ? '6px 16px' : '10px 18px',
    fontWeight: isSelected && isGroup(options) ? 600 : 400,
    ...(isGroup(options) && {
      'div > svg': {
        fill: theme.general.contrastColor,
      },
    }),
    ...(!isGroup(options) && {
      '&:not(:last-of-type)': {
        borderBottom: `1px solid ${theme.Select.borderColor}`,
      },
    }),
    ...(!isGroup(options) && {
      '&:first-of-type': {
        borderRadius: '0',
      },
    }),
    ...(isGroup(options) && {
      textAlign: 'left',
    }),
    '&:last-of-type': {
      ...(!isGroup(options) && {borderRadius: '0'}),
      ...(isGroup(options) &&
        !options[options.length - 1].options.includes(data) && {
          borderBottom: `1px solid ${theme.Select.borderColor}`,
        }),
    },
  }),
  valueContainer: provided => ({
    ...provided,
    padding: 0,
    width: '100%',
    display: 'grid',
  }),
  groupHeading: base => ({
    ...base,
    color: theme.general.color,
    fontSize: theme.fontSizes.small,
    lineHeight: 1,
    padding: '8px 16px 6px',
    textAlign: 'left',
    ...groupHeadingCSS,
  }),
  group: provided => ({
    ...provided,
    backgroundColor: theme.Input.backgroundColor,
    ...groupCSS,
  }),
});
