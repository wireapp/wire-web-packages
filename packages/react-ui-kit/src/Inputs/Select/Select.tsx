/*
 * Wire
 * Copyright (C) 2018 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import {ReactElement} from 'react';

import {CSSObject, useTheme} from '@emotion/react';
import ReactSelect, {GroupBase, StylesConfig} from 'react-select';
import type {StateManagerProps} from 'react-select/dist/declarations/src/stateManager';

import {BaseSelectDropdownIndicator} from './BaseSelect/BaseSelectDropdownIndicator';
import {customStyles} from './Select.styles';
import {SelectContainer} from './SelectContainer/SelectContainer';
import {SelectIndicatorsContainer} from './SelectIndicatorsContainer/SelectIndicatorsContainer';
import {SelectMenu} from './SelectMenu/SelectMenu';
import {SelectMenuList} from './SelectMenuList/SelectMenuList';
import {isGroup, SelectOption} from './SelectOption/SelectOption';
import {SelectValueContainer} from './SelectValueContainer/SelectValueContainer';

import {Theme} from '../../Identity/Theme';
import {TabIndex} from '../../utils/enums';
import {InputLabel} from '../InputLabel';

export type Option = {
  value: string | number;
  label: string;
  description?: string;
  isDisabled?: boolean;
};

export interface SelectProps<IsMulti extends boolean, Group extends GroupBase<Option>>
  extends StateManagerProps<Option, IsMulti, Group> {
  id: string;
  disabled?: boolean;
  dataUieName: string;
  options: Option[] | Group[];
  menuCSS?: CSSObject;
  wrapperCSS?: CSSObject;
  selectControlCSS?: CSSObject;
  selectContainerCSS?: CSSObject;
  label?: string;
  helperText?: string;
  error?: ReactElement;
  markInvalid?: boolean;
  required?: boolean;
  isMulti?: IsMulti;
  isSearchable?: boolean;
  overlayMenu?: boolean;
  menuListHeading?: string;
}

export const Select = <IsMulti extends boolean = false, Group extends GroupBase<Option> = GroupBase<Option>>({
  id,
  label,
  error,
  helperText,
  disabled = false,
  dataUieName,
  options,
  isMulti,
  wrapperCSS = {},
  menuCSS = {},
  selectControlCSS = {},
  selectContainerCSS = {},
  markInvalid = false,
  required = false,
  isSearchable = false,
  overlayMenu = true,
  menuListHeading,
  ...props
}: SelectProps<IsMulti, Group>) => {
  const theme = useTheme();
  const hasError = !!error;

  return (
    <div
      // eslint-disable-next-line jsx-a11y/no-autofocus
      autoFocus={isGroup(options)}
      css={(theme: Theme) => ({
        marginBottom: markInvalid ? '2px' : '20px',
        width: '100%',
        '&:focus-within label': {
          color: theme.general.primaryColor,
        },
        ...wrapperCSS,
      })}
      data-uie-name={dataUieName}
    >
      {label && (
        <InputLabel htmlFor={id} markInvalid={markInvalid} isRequired={required}>
          {label}
        </InputLabel>
      )}

      <ReactSelect
        id={id}
        styles={
          customStyles({
            theme: theme as Theme,
            markInvalid,
            menuPosition: overlayMenu ? 'absolute' : 'relative',
            controlCSS: selectControlCSS,
            containerCSS: selectContainerCSS,
          }) as StylesConfig
        }
        components={{
          SelectContainer: SelectContainer,
          DropdownIndicator: BaseSelectDropdownIndicator,
          Option: SelectOption(dataUieName),
          Menu: SelectMenu(dataUieName, menuCSS),
          ValueContainer: SelectValueContainer,
          IndicatorsContainer: SelectIndicatorsContainer,
          ...(menuListHeading && {MenuList: SelectMenuList(menuListHeading, dataUieName)}),
        }}
        tabIndex={TabIndex.UNFOCUSABLE}
        isDisabled={disabled}
        hideSelectedOptions={false}
        isSearchable={isSearchable}
        isClearable={false}
        closeMenuOnSelect={!isMulti}
        isMulti={isMulti}
        options={options}
        {...props}
      />

      {!hasError && helperText && (
        <p
          css={(theme: Theme) => ({
            fontSize: theme.fontSizes.small,
            fontWeight: 400,
            color: theme.Input.labelColor,
            marginTop: 8,
          })}
        >
          {helperText}
        </p>
      )}

      {error}
    </div>
  );
};
