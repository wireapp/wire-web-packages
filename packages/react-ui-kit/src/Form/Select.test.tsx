/*
 * Wire
 * Copyright (C) 2019 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

/* eslint-disable jest/expect-expect */

import * as React from 'react';

import {Select} from './Select';

import {matchComponent} from '../test/testUtil';

const props: React.ComponentProps<typeof Select> = {
  options: [],
  id: 'test',
  dataUieName: 'test',
};

describe('"Select"', () => {
  it('renders', () =>
    matchComponent(
      <Select {...props}>
        <option>a</option>
        <option>b</option>
      </Select>,
    ));
  it('renders as disabled', () => matchComponent(<Select {...props} disabled></Select>));
  it('renders as invalid', () => matchComponent(<Select {...props} markInvalid></Select>));
});
