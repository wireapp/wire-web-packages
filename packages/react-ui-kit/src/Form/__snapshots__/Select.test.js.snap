// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"Select" renders 1`] = `
<div
  className="css-7uml9n-StyledAppContainer"
>
  <div
    className="css-2ocyf0-Select"
  >
    <div
      className=" css-dkntwv-SelectContainer"
      onKeyDown={[Function]}
    >
      <div
        tabIndex={0}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-2-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className=" css-6smaub-Control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className=" css-16ieua5-ValueContainer"
          >
            <div
              className=" css-14el2xx-placeholder"
              id="react-select-2-placeholder"
            >
              Select...
            </div>
             
            <input
              aria-autocomplete="list"
              aria-describedby="react-select-2-placeholder"
              aria-expanded={false}
              aria-haspopup={true}
              aria-readonly={true}
              className="css-mohuvp-dummyInput-DummyInput"
              disabled={false}
              id="react-select-2-input"
              inputMode="none"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              role="combobox"
              tabIndex={-1}
              value=""
            />
          </div>
          <div
            className=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              className=" css-1xjgjl1-IndicatorSeparator"
            />
            <div
              aria-hidden="true"
              className=" css-1186e7r-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <svg
                className="css-1tuya7p-DropdownIndicator"
                height={16}
                viewBox="0 0 16 16"
                width={16}
              >
                <g>
                  <path
                    d="M7.99963 12.5711L15.6565 4.91421L14.2423 3.5L7.99963 9.74264L1.75699 3.5L0.342773 4.91421L7.99963 12.5711Z"
                  />
                </g>
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`"Select" renders as disabled 1`] = `
<div
  className="css-7uml9n-StyledAppContainer"
>
  <div
    className="css-2ocyf0-Select"
  >
    <div
      className=" css-19vi2h5-SelectContainer"
      onKeyDown={[Function]}
    >
      <div
        tabIndex={0}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-3-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className=" css-6smaub-Control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className=" css-16ieua5-ValueContainer"
          >
            <div
              className=" css-14el2xx-placeholder"
              id="react-select-3-placeholder"
            >
              Select...
            </div>
             
            <input
              aria-autocomplete="list"
              aria-describedby="react-select-3-placeholder"
              aria-expanded={false}
              aria-haspopup={true}
              aria-readonly={true}
              className="css-mohuvp-dummyInput-DummyInput"
              disabled={true}
              id="react-select-3-input"
              inputMode="none"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              role="combobox"
              tabIndex={-1}
              value=""
            />
          </div>
          <div
            className=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              className=" css-1xjgjl1-IndicatorSeparator"
            />
            <div
              aria-hidden="true"
              className=" css-zbce0a-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <svg
                className="css-1tuya7p-DropdownIndicator"
                height={16}
                viewBox="0 0 16 16"
                width={16}
              >
                <g>
                  <path
                    d="M7.99963 12.5711L15.6565 4.91421L14.2423 3.5L7.99963 9.74264L1.75699 3.5L0.342773 4.91421L7.99963 12.5711Z"
                  />
                </g>
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`"Select" renders as invalid 1`] = `
<div
  className="css-7uml9n-StyledAppContainer"
>
  <div
    className="css-yrzhnj-Select"
  >
    <div
      className=" css-g607jh-SelectContainer"
      onKeyDown={[Function]}
    >
      <div
        tabIndex={0}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-4-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className=" css-6smaub-Control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className=" css-16ieua5-ValueContainer"
          >
            <div
              className=" css-14el2xx-placeholder"
              id="react-select-4-placeholder"
            >
              Select...
            </div>
             
            <input
              aria-autocomplete="list"
              aria-describedby="react-select-4-placeholder"
              aria-expanded={false}
              aria-haspopup={true}
              aria-readonly={true}
              className="css-mohuvp-dummyInput-DummyInput"
              disabled={false}
              id="react-select-4-input"
              inputMode="none"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              role="combobox"
              tabIndex={-1}
              value=""
            />
          </div>
          <div
            className=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              className=" css-1xjgjl1-IndicatorSeparator"
            />
            <div
              aria-hidden="true"
              className=" css-1186e7r-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <svg
                className="css-1tuya7p-DropdownIndicator"
                height={16}
                viewBox="0 0 16 16"
                width={16}
              >
                <g>
                  <path
                    d="M7.99963 12.5711L15.6565 4.91421L14.2423 3.5L7.99963 9.74264L1.75699 3.5L0.342773 4.91421L7.99963 12.5711Z"
                  />
                </g>
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
